scvi/dataset/anndata.py

```diff
@@ -0,0 +1,35 @@
+from .dataset import GeneExpressionDataset
+import anndata
+import os
+import zipfile
+import numpy as np
+
+
+class AnnDataset(GeneExpressionDataset):
+
+    def __init__(self, filename, download_name=None, save_path='data/', url=None):
+        self.download_name = download_name if download_name is not None else filename
+        self.filename = filename
+        self.save_path = save_path
+        self.url = url
+
+        data, gene_names = self.download_and_preprocess()
+        print("Finished preprocessing dataset")
+
+        super(AnnDataset, self).__init__(*GeneExpressionDataset.get_attributes_from_matrix(data),
+                                         gene_names=gene_names)
+
+    def preprocess(self):
+        print("Preprocessing data")
+        if not os.path.isfile(self.save_path + self.filename):  # nothing extracted yet
+            with zipfile.ZipFile(self.save_path + self.download_name) as zf:
```

Comment:
Better not to support downloading zip files, I think. Just support (unzipped) .h5ad files on disk, and urls that directly reference .h5ad files. (Even though we don't have any examples of the latter)